src/nvme/fabrics.c

```diff
@@ -838,6 +837,64 @@ static int uuid_from_dmi(char *system_uuid)
 	return strlen(system_uuid) ? 0 : -ENXIO;
 }
 
+#define PATH_DMI_PROD_UUID  "/sys/class/dmi/id/product_uuid"
+static int uuid_from_product_uuid(char *system_uuid)
+{
+	FILE *stream = NULL;
+	int   ret    = -ENXIO;
+
+	system_uuid[0] = '\0';
+
+	if ((stream = fopen(PATH_DMI_PROD_UUID, "re")) != NULL) {
+		char    *line  = NULL;
+		size_t   len   = 0;
+		ssize_t  nread = getline(&line, &len, stream);
+
+		if (nread == UUID_SIZE) {
+			static struct { int from; int to; } swaptbl[] = {
+				{0, 6}, {1, 7}, {2, 4}, {3, 5}, {9, 11}, 
+				{10, 12}, {14, 16}, {15, 17}, {-1, -1}
+			};
+			unsigned int i;
+			char tmp;
+
+			memcpy(system_uuid, line, UUID_SIZE);
+			system_uuid[UUID_SIZE-1]= '\0'; /* \n -> \0 */
+
+			/* Swap little-endian to network order per
+			   DMTF SMBIOS 3.0 Section 7.2.1 System UUID */
+			for (i = 0; swaptbl[i].from != -1; i++) {
+				tmp = system_uuid[swaptbl[i].from];
+				system_uuid[swaptbl[i].from] = system_uuid[swaptbl[i].to];
+				system_uuid[swaptbl[i].to] = tmp;
```

Comment:
Nice! Could you make uuid_from_dmi_entries() also using this swap code? Maybe have it as small helper function?

Reply:
Ok, makes sense.